quickwit-indexing/src/merge_policy.rs

```diff
@@ -204,19 +208,74 @@ impl StableMultitenantWithTimestampMergePolicy {
 
         let mut split_levels: Vec<Range<usize>> = Vec::new();
         let mut current_level_start_ord = 0;
-        let mut current_level_max_docs = (splits[0].num_records * 3).max(self.min_level_num_docs);
+        let mut current_level_max_docs =
+            (splits[0].num_records * 3).clamp(self.min_level_num_docs, self.max_merge_docs);
 
         for (split_ord, split) in splits.iter().enumerate() {
             if split.num_records >= current_level_max_docs {
                 split_levels.push(current_level_start_ord..split_ord);
                 current_level_start_ord = split_ord;
-                current_level_max_docs = 3 * split.num_records;
+                current_level_max_docs = (3 * split.num_records).min(self.max_merge_docs);
+                if split.num_records >= self.max_merge_docs {
+                    return split_levels;
+                }
```

Comment:
this early return sounds very strange and the code will never reach this part, right?

Reply:
ah you are absolutely right. It is the leftover of a failed refactorign attempt.